<!-- --> 
<launch>

	<node name="optitrack" pkg="sim_optitrack" type="sim_optitrack_node" output="screen">
			<rosparam>rigid_bodies_name:  [quadrotor,leader]</rosparam>
	</node>

	<node name="drone_target_points" pkg="drone_following_leader" type="following_leader_node" output="screen">
		<remap from="leader_pose_topic"  to="/optitrack/rigid_bodies"/>
		<remap from="leader_info_topic"  to="/drone_target_points"/>
	</node>

	<node name="drone_control_node" pkg="drone_control_position" type="sim_control_drone_position_node" output="screen">
		<remap from="drone_info_topic"  to="/optitrack/rigid_bodies"/>
		<remap from="control_info"  to="/drone_target_points"/>
		<rosparam>Kd:  [0.5, 0.5]</rosparam>  <!-- [pitch, roll] -->
		<rosparam>Kp:  [0.5, 0.5, 0.25, 0.05]</rosparam>  <!-- [pitch, roll, altitude, yaw] -->
		<rosparam>velocity_limit:  [1.0, 1.0, 1.0, 1.0]</rosparam>  <!-- [pitch, roll, altitude, yaw] -->
		<rosparam>delta_pose:  [-1.0, 0.0, 0.0, 0.0]</rosparam>  <!-- [x,y,z,yaw (degrees)] -->
		<rosparam>virtual_fence:  [10.0, -10.0, 10.0, -10.0, 2.5]</rosparam>  <!-- [maxX, minX, maxY, minY, maxAltitude] -->
	</node>

</launch>
